src/docfx/validation/ContentValidator.cs

```diff
@@ -49,7 +51,10 @@ internal class ContentValidator : ICollectionFactory
                 fileResolver.ResolveFilePath(_config.MarkdownValidationRules),
                 fileResolver.ResolveFilePath(_config.Allowlists),
                 fileResolver.ResolveFilePath(_config.SandboxEnabledModuleList),
-                this);
+                this,
+                EnvironmentVariable.GetDocsEnvironment().ToString(),
```

Comment:
Why validator needs to understand environment?

Reply:
the env is passing to broken link service eventually, and broken link service needs it to decide which redis cache to hit.

Reply:
I see, but would it be better that we pass in environment setting from DocFX to Broken Link library? This way it might be easier for us to adjust the setting.

Reply:
Chatted with Li offline, I am okay to accept passing in an environment name.